src/main/resources/templates/settingsForSingleProject.vm

```diff
@@ -88,19 +88,19 @@ $webResourceManager.requireResource("de.uhd.ifi.se.decision.management.jira:styl
 		}
 		</script>
 		<script>
+		$(document).ready(function () {
 		for (j in additionalKnowledgeTypes) {
-			console.log(typeToggles[j].value);
-			isKnowledgeTypeEnabled(typeToggles[j].value, '$projectKey', typeToggles[j], function(isTypeEnabled, toggle) {				
-                toggle.checked = isTypeEnabled;
-           	});
-            $(document).ready(function () {
-                typeToggles[j].addEventListener('change', function (error) {
-                    this.busy = true;
-                    setKnowledgeTypeEnabled(this.checked, this.value, '$projectKey');
-                    this.busy = false;
-                });
-            });
-		}
+				console.log(typeToggles[j].value);
+				isKnowledgeTypeEnabled(typeToggles[j].value, '$projectKey', typeToggles[j], function(isTypeEnabled, toggle) {				
+                	toggle.checked = isTypeEnabled;
+           		});
+                	typeToggles[j].addEventListener('change', function (error) {
+                    	this.busy = true;
+                    	setKnowledgeTypeEnabled(this.checked, this.value, '$projectKey');
+                    	this.busy = false;
+                	});           
+				}
+			});
```

Comment:
This is better, but ideally settingsForSingleProject.vm should have its own javascript file or at most one script tag in it with functions.

Then only one appearence of `$(document).ready`  should be used, and within its block functions could be called, for example:
```
$(document).ready(function () { 
 settingsForSingleProjectController.setKnowledgeTypeToggles();
 ... do other stuff ...
}

```

Reply:
I moved the easy to move code at the end of the file. The remaining Javascript parts are not working when they are moved.